src/Containers/DenseAxisArray.jl

```diff
@@ -674,7 +734,34 @@ end
 
 _type_stable_args(axes::Tuple, ::Tuple{}) = axes
 
-function Base.getindex(x::DenseAxisArrayView, args...)
+function _fixed_indices(view_axes::Tuple, axes::Tuple)
+    return filter(ntuple(i -> i, length(view_axes))) do i
+        return !(typeof(view_axes[i]) <: eltype(axes[i]))
+    end
+end
+
+function _kwargs_to_args(A::DenseAxisArrayView{T,N}; kwargs...) where {T,N}
+    non_default_indices = _fixed_indices(A.axes, A.data.axes)
+    return ntuple(N) do i
+        kw = keys(kwargs)[i]
+        if A.data.names[non_default_indices[i]] != kw
+            error(
+                "Invalid index $kw in position $i. When using keyword " *
+                "indexing, the indices must match the exact name and order " *
+                "used when creating the container.",
+            )
+        end
+        return kwargs[i]
+    end
+end
+
+function Base.getindex(x::DenseAxisArrayView, args...; kwargs...)
+    if !isempty(kwargs)
```

Comment:
`getindex` is performance sensitive. Is there any impact from this extra `!isempty(kwargs)` in the case of all positional arguments?

Reply:
Nope. It gets compiled away:

```julia
shell> git status
On branch od/container-kwarg-getindex
Your branch is up to date with 'origin/od/container-kwarg-getindex'.

nothing to commit, working tree clean

julia> using JuMP

julia> using BenchmarkTools

julia> model = Model()
A JuMP Model
Feasibility problem with:
Variables: 0
Model mode: AUTOMATIC
CachingOptimizer state: NO_OPTIMIZER
Solver name: No optimizer attached.

julia> @variable(model, x[i=2:3, j=4:5])
2-dimensional DenseAxisArray{VariableRef,2,...} with index sets:
    Dimension 1, 2:3
    Dimension 2, 4:5
And data, a 2×2 Matrix{VariableRef}:
 x[2,4]  x[2,5]
 x[3,4]  x[3,5]

julia> @btime getindex($x, 2, 4)
  3.466 ns (0 allocations: 0 bytes)
x[2,4]

julia> exit()
(base) oscar@Oscars-MBP JuMP % git checkout master
Switched to branch 'master'
Your branch is up to date with 'origin/master'.
(base) oscar@Oscars-MBP JuMP % julia --project=.  
               _
   _       _ _(_)_     |  Documentation: https://docs.julialang.org
  (_)     | (_) (_)    |
   _ _   _| |_  __ _   |  Type "?" for help, "]?" for Pkg help.
  | | | | | | |/ _` |  |
  | | |_| | | | (_| |  |  Version 1.6.7 (2022-07-19)
 _/ |\__'_|_|_|\__'_|  |  Official https://julialang.org/ release
|__/                   |

julia> using JuMP
[ Info: Precompiling JuMP [4076af6c-e467-56ae-b986-b466b2749572]

julia> using BenchmarkTools

julia> model = Model()
A JuMP Model
Feasibility problem with:
Variables: 0
Model mode: AUTOMATIC
CachingOptimizer state: NO_OPTIMIZER
Solver name: No optimizer attached.

julia> @variable(model, x[i=2:3, j=4:5])
2-dimensional DenseAxisArray{VariableRef,2,...} with index sets:
    Dimension 1, 2:3
    Dimension 2, 4:5
And data, a 2×2 Matrix{VariableRef}:
 x[2,4]  x[2,5]
 x[3,4]  x[3,5]

julia> @btime getindex($x, 2, 4)
  3.725 ns (0 allocations: 0 bytes)
x[2,4]
```

There's still room to improve the `kwarg` option, but let's settle on the syntax/opt-in stuff first:
```julia
julia> @btime getindex($x; i = 2, j = 4)
  33.646 ns (1 allocation: 32 bytes)
x[2,4]
```

Reply:
Is kwarg indexing fundamentally 10x slower than positional indexing? If that's true we might want to reconsider the syntax.

Reply:
No, if you change the method to a singular `getindex(x; kwargs...)` it comes down to 12ns and 0 allocations, but then I needed to add a bunch of additional methods to avoid ambiguities. Left as-is for now while we sort the syntax.

Keyword indexing in general does have a slight performance hit though.

Reply:
> Keyword indexing in general does have a slight performance hit though.

In fact, keyword indexing can be as fast as Cartesian indexing if the tuple of names is encoded into the customized array type, but it is obviously not an option for JuMP now.